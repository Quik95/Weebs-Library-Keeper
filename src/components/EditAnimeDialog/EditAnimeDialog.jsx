import React from "react";
import PropTypes from "prop-types";

// MUI
import Dialog from "@material-ui/core/Dialog";

// Custom components
import DialogTitle from "./DialogTitle/DialogTitle";
import DialogActions from "./DialogActions/DialogActions";
import DialogContent from "./DialogContent/DialogContent";

export default function EditAnimeDialog({ maxWidth, open, handleClose }) {
  return (
    <Dialog
      maxWidth={maxWidth}
      fullWidth
      open={open}
      onClose={handleClose}
      aria-labelledby="edit-anime-dialog"
    >
      <DialogTitle onClose={handleClose}>Dr. Stone</DialogTitle>
      <DialogContent animeId="to-do" />
      <DialogActions handleClose={handleClose} />
    </Dialog>
  );
}

EditAnimeDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  maxWidth: PropTypes.string.isRequired
};
