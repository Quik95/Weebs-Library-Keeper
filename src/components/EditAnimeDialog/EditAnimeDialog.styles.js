import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    margin: 'auto',
    width: 'fit-content',
  },
  formControl: {
    marginTop: theme.spacing(2),
    minWidth: 120,
  },
  formControlLabel: {
    marginTop: theme.spacing(1),
  },
  dialogTitle: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
}));
