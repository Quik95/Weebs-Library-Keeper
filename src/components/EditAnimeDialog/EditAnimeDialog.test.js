import EditAnimeDialog from './EditAnimeDialog';

describe('<EditAnimeDialog/>', () => {
  const onCloseMock = jest.fn(() => {});

  const wrapper = shallow(
    <EditAnimeDialog maxWidth="md" open handleClose={onCloseMock} />
  );

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
