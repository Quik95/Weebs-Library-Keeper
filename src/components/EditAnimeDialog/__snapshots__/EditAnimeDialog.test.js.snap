// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditAnimeDialog/> Renders correctly 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="edit-anime-dialog"
  fullWidth={true}
  maxWidth="md"
  onClose={[MockFunction]}
  open={true}
>
  <Memo(DialogTitle)
    onClose={[MockFunction]}
  >
    Dr. Stone
  </Memo(DialogTitle)>
  <DialogContent
    animeId="to-do"
  />
  <Memo(DialogActions)
    handleClose={[MockFunction]}
  />
</WithStyles(ForwardRef(Dialog))>
`;
